### PR TITLE
perf: Use method directly in onclicks instead of creating new one on each render

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

**Comment by ChefMomota** [comment]
LGTM

File: src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx [hunk 1]
```diff
@@ -69,9 +69,7 @@ const EditNumbersModal: React.FC<{
           id="lotteryBuyEdited"
           disabled={!allComplete || isConfirming}
           endIcon={isConfirming ? <AutoRenewIcon spin color="currentColor" /> : undefined}
-          onClick={() => {
-            onConfirm()
-          }}
+          onClick={onConfirm}
```
Comment (chefilip) [inline src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx:72]: For this onClick event, I think we should avoid passing `event` param to the callback.

Comment (memoyil) [reply src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx:72]: Done

File: src/views/LimitOrders/components/SwitchTokensButton.tsx [hunk 2]
```diff
@@ -28,16 +28,12 @@ interface SwitchIconButtonProps {
...
     <AutoColumn justify="space-between">
       <AutoRow justify="center" style={{ padding: '0 1rem' }}>
-        <StyledButton
-          variant="light"
-          scale="sm"
-          onClick={() => {
-            handleSwitchTokens()
-          }}
-        >
+        <StyledButton variant="light" scale="sm" onClick={handleOnClick}>
```
Comment (chef-jojo) [inline src/views/LimitOrders/components/SwitchTokensButton.tsx:36]: onClick={handleSwitchTokens}

File: src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx [hunk 3]
```diff
@@ -69,9 +71,7 @@ const EditNumbersModal: React.FC<{
           id="lotteryBuyEdited"
           disabled={!allComplete || isConfirming}
           endIcon={isConfirming ? <AutoRenewIcon spin color="currentColor" /> : undefined}
-          onClick={() => {
-            onConfirm()
-          }}
+          onClick={handleOnConfirm}
```
Comment (chef-jojo) [inline src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx:74]: onClick={onConfirm}

File: src/components/ExpandableSectionButton/ExpandableSectionButton.tsx [hunk 4]
```diff
@@ -20,9 +21,10 @@ const Wrapper = styled.div`
 
 const ExpandableSectionButton: React.FC<ExpandableSectionButtonProps> = ({ onClick, expanded = false }) => {
   const { t } = useTranslation()
+  const handleOnClick = useCallback(() => onClick?.(), [onClick])
 
   return (
-    <Wrapper aria-label={t('Hide or show expandable content')} role="button" onClick={() => onClick()}>
+    <Wrapper aria-label={t('Hide or show expandable content')} role="button" onClick={handleOnClick}>
```
[the remark this answers is not in the document]
Comment (chef-jojo) [reply src/components/ExpandableSectionButton/ExpandableSectionButton.tsx:27]: Umm i think it's okay. Anyway no biggie

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**